x-pack/plugins/security/public/account_management/user_profile/utils.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export const IMAGE_FILE_TYPES = ['image/svg+xml', 'image/jpeg', 'image/png', 'image/gif'];
+export const MAX_IMAGE_SIZE = 64;
+
+export function readFile(data: File) {
+  return new Promise<string>((resolve, reject) => {
+    const reader = new FileReader();
+    reader.onloadend = () => resolve(reader.result as string);
+    reader.onerror = reject;
+    reader.readAsDataURL(data);
+  });
+}
+
+export function resizeImage(imageUrl: string, maxSize: number) {
+  return new Promise<string>((resolve, reject) => {
+    const image = new Image();
+    image.onload = () => {
+      if (image.width <= maxSize && image.height <= maxSize) {
+        return resolve(imageUrl);
+      }
+
+      try {
+        const canvas = document.createElement('canvas');
+        const context = canvas.getContext('2d');
+        if (context) {
+          if (image.width >= image.height) {
+            canvas.width = maxSize;
+            canvas.height = Math.floor((image.height * maxSize) / image.width);
+          } else {
+            canvas.height = maxSize;
+            canvas.width = Math.floor((image.width * maxSize) / image.height);
+          }
+          context.drawImage(image, 0, 0, canvas.width, canvas.height);
+          const resizedDataUrl = canvas.toDataURL();
```

Comment:
Funnily enough, centering "tall" images works much worse for the image set I have since the head in tall photos is usually close to (0,0) and centering clips the head sometimes like in the picture below:

![image](https://user-images.githubusercontent.com/1713708/172579372-1adb51ea-d6d1-47c4-ad77-0543ac7b05f5.png)

I'll keep as is for now, and we can revisit it with @thomheymann once he's back